### DISCUSSION TITLE
SSH login requires "enable SSO" which does not exist

### DISCUSSION THREAD

**Discussion by graemev** [community/community#23893] [New to GitHub]
<p>I just know this going to get me flamed (with RTFM) but I just created a githib account. I want to push my repo to it so I used the URL</p>
<p>git remote add origin <a href="https://github.com/" rel="noopener nofollow ugc">https://github.com/</a>/<br>
Got password prompt …several steps later …</p>
<blockquote>
<p>Beginning August 13, 2021, we will no longer accept account passwords when authenticating Git operations</p>
</blockquote>
<p>So looked at setting up SSH access. Added my public key. Manage to get through the test</p>
<p>(here but I’m not allowed to post URL)   /en/github/authenticating-to-github/connecting-to-github-with-ssh/testing-your-ssh-connection</p>
<p>ie:  $ ssh -T <a href="mailto:git@github.com">git@github.com</a></p>
<p>So back to my original</p>
<blockquote>
<p>git push --set-upstream origin master</p>
</blockquote>
<p>But again prompted for password, so RTFM</p>
<p>(here but I’m not allowed to post URL)/en/github/authenticating-to-github/authenticating-with-saml-single-sign-on/authorizing-an-ssh-key-for-use-with-saml-single-sign-on</p>
<p>and it says … “Enable SSO”  …but there is no such button.</p>
<p>Have passwords been disabled but the replacement feature not enabled for mortal users?</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="graemev" data-post="1" data-topic="197874">
<div class="title">
<div class="quote-controls"></div>
 graemev:</div>
<blockquote>
<p>But again prompted for password, so RTFM</p>
</blockquote>
</aside>
<p>That sounds like the repository URL still uses HTTPS, not SSH. You can check the URL with:</p>
<pre><code class="lang-auto">git remote show origin
</code></pre>
<p>If there are still <code>https://github.com/...</code> URLs, you can change to SSH with the following command. Replace the URL with that for your repository, the pattern should be the same.</p>
<pre><code class="lang-nohighlight">git remote set-url origin git@github.com:user/repo.git
</code></pre>

**Comment by graemev** [comment]
<p>Thank you it was indeed that. I discovered by battering my head against it repeatedly .</p>
<p>The user experience was very poor. I setup the URL as per instructions, then got a message saying that “password access” had been suspended (as a method) and I had to use SSH or some kind of access token.  Very happy to use SSH (preferable in fact)</p>
<p>A better approach would have been, "We no longer support xxxx (e.g. password authentication) you need to switch to methodA or methodB</p>
<p>+guide to setting up methodA,<br>
+guide to switching from current-method to methodA</p>
<p>ie it was perfectly possible they could have used HTTPS but required me to authenticate using SSH . On reflection that way they’ve done it is more sensible/standard …I just wish they’d told me!</p>
<p>Anyhow thanks for your help. Hopefully it will help future users (with that in mind, I’ll add some keywords)</p>
<p>password access, ssh access, SSH login, git-URL</p>
